session.go

```diff
@@ -227,18 +227,46 @@ func (s *Session) init() error {
 	}
 
 	hosts = hosts[:0]
+	// each host will increment left and decrement it after connecting and once
+	// there's none left, we'll close hostCh
+	var left int64
+	// we will receive up to len(hostMap) of messages so create a buffer so we
+	// don't end up stuck in a goroutine if we stopped listening
+	connectedCh := make(chan struct{}, len(hostMap))
+	// we add one here because we don't want to end up closing hostCh until we're
+	// done looping and the decerement code might be reached before we've looped
+	// again
+	atomic.AddInt64(&left, 1)
 	for _, host := range hostMap {
-		host = s.ring.addOrUpdate(host)
+		host := s.ring.addOrUpdate(host)
 		if s.cfg.filterHost(host) {
 			continue
 		}
 
-		host.setState(NodeUp)
-		s.pool.addHost(host)
+		atomic.AddInt64(&left, 1)
+		go func() {
+			host.setState(NodeUp)
+			s.pool.addHost(host)
+			s.policy.HostUp(host)
```

Comment:
I added this here because during testing, nothing was calling `HostUp` so the ReadyPolicy wasn't working.

Reply:
Maybe in `session.addNewNode` we should instead call `policy.HostUp` instead of `policy.AddHost`?